server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java

```diff
@@ -217,15 +252,25 @@ public Builder setStartedAtMillis(long startedAtMillis) {
             return this;
         }
 
+        public Builder setShardReallocationDelay(TimeValue shardReallocationDelay) {
+            this.shardReallocationDelay = shardReallocationDelay;
+            return this;
+        }
+
         public SingleNodeShutdownMetadata build() {
             if (startedAtMillis == -1) {
                 throw new IllegalArgumentException("start timestamp must be set");
             }
+            TimeValue delayOrDefault = shardReallocationDelay;
+            if (Type.RESTART.equals(type) && delayOrDefault == null) {
+                delayOrDefault = DEFAULT_RESTART_SHARD_ALLOCATION_DELAY;
```

Comment:
This feels like the wrong place to implement this default, I think rather than here, it should go into the getter inside of `SingleNodeShutdownMetadata`, otherwise calling `toXContent` on the metadata makes it appear that it has been set (when in reality the default value will be used)

Reply:
Good call, I'll change it.